ICSharpCode.Decompiler/Disassembler/MethodBodyDisassembler.cs

```diff
@@ -66,9 +66,9 @@ public void Disassemble(MethodBody body, MethodDebugSymbols debugSymbols)
 				foreach (var v in method.Body.Variables) {
 					output.WriteDefinition("[" + v.Index + "] ", v);
 					v.VariableType.WriteTo(output);
-					if (!string.IsNullOrEmpty(v.Name)) {
+					if (!string.IsNullOrEmpty(v.ToString())) {
```

Comment:
On a similar note to the stuff above, see [here](https://github.com/airbreather/ILSpy/commit/4d18cd33e29de480f172b0b23aa4a1fefab845fa#diff-ae2cfc63bf9b110e4808363d83385b72R69) for a tweak to get the variable's proper name from debug information at this point in the code.